services/data/src/react/hooks/useDataMutation.ts

```diff
@@ -1,32 +1,79 @@
-import { useCallback } from 'react'
-import { QueryOptions, Mutation } from '../../engine'
+import { useState, useCallback } from 'react'
+import { useMutation } from 'react-query'
+import { QueryVariables, QueryOptions, Mutation } from '../../engine'
 import { MutationRenderInput } from '../../types'
 import { useDataEngine } from './useDataEngine'
-import { useQueryExecutor } from './useQueryExecutor'
 import { useStaticInput } from './useStaticInput'
 
-const empty = {}
 export const useDataMutation = (
     mutation: Mutation,
-    { onComplete, onError, variables = empty, lazy = true }: QueryOptions = {}
+    {
+        onComplete: userOnComplete,
+        onError: userOnError,
+        variables: initialVariables = {},
+        lazy = true,
+    }: QueryOptions = {}
 ): MutationRenderInput => {
     const engine = useDataEngine()
+    const [variables, setVariables] = useState(initialVariables)
+    const [shouldMutateNow, setShouldMutateNow] = useState(!lazy)
     const [theMutation] = useStaticInput<Mutation>(mutation, {
```

Comment:
```suggestion
    const [resolvedMutationQuery] = useStaticInput<Mutation>(mutation, {
```

Reply:
Resolved in 655f42633e777e54688c884cdcda8199285af991

Reply:
Resolved in 655f42633e777e54688c884cdcda8199285af991